lib/hanami/assets/helpers.rb

```diff
@@ -186,9 +240,25 @@ def javascript(*sources)
       #   <%= stylesheet 'application' %>
       #
       #   # <link href="https://assets.bookshelf.org/assets/application-28a6b886de2372ee3922fcaf3f78f2d8.css" type="text/css" rel="stylesheet">
-      def stylesheet(*sources)
+      def stylesheet(*sources, **options)
         _safe_tags(*sources) do |source|
-          html.link(href: _typed_asset_path(source, STYLESHEET_EXT), type: STYLESHEET_MIME_TYPE, rel: STYLESHEET_REL).to_s
+          tag_options = options.merge({
```

Comment:
@accuser 

> For example, it wouldn't make sense for `href` to be specified when multiple sources were provided, for example.

That is the reason why these options weren't included in the initial implementation. It's handy to have a signature that allows to include multiples assets at once:

```erb
<%= javascript 'jquery', 'application' %>
```

...but a bit awkward to specify single options:

```erb
<%= javascript 'jquery', 'application', id: 'myscript' %>
```

This would generate invalid markup because both the scripts will receive the same DOM ID.

---

We're at a forking path:

  1. Allow developers to use `javascript` to specify one or more scripts, by leaving to them the responsibility of not making mistakes.

  2. Be strict and make `javascript` to generate only one `<script>` tag at once.

I'm for 1 and document this behavior in guides.

---

In any case we should allow them to force `href`, `type` and `rel` attributes. We can never tell which needs a developer can have, we should work to second their needs in this case.

It would be odd to write the following code and get something unexpected in return:

```erb
<%= javascript 'entry-template', type: 'text/x-handlebars-template' %>
```